stl/inc/functional

```diff
@@ -1207,6 +1207,694 @@ _NODISCARD bool operator!=(nullptr_t, const function<_Fty>& _Other) noexcept {
 }
 #endif // !_HAS_CXX20
 
+#if _HAS_CXX23
+union alignas(max_align_t) _Move_only_function_data {
+    void* _Pointers[_Small_object_num_ptrs];
+    const void* _Impl;
+    char _Data; // For aliasing
+
+    template <class _Fn>
+    static constexpr size_t _Buf_offset = alignof(_Fn) <= sizeof(_Impl)
+                                            ? sizeof(_Impl) // Store _Fn immediately after _Impl
+                                            : alignof(_Fn); // Pad _Fn to next alignment
+
+    template <class _Fn>
+    static constexpr size_t _Buf_size = sizeof(_Pointers) - _Buf_offset<_Fn>;
+
+    template <class _Fn>
+    _NODISCARD void* _Buf_ptr() noexcept {
+        return &_Data + _Buf_offset<_Fn>;
+    }
+
+    template <class _Fn>
+    _NODISCARD _Fn* _Small_fn_ptr() const noexcept {
+        // cast away const to avoid complication of const propagation to here;
+        // const correctness is still enforced by _Move_only_function_call specializations.
+        return static_cast<_Fn*>(const_cast<_Move_only_function_data*>(this)->_Buf_ptr<_Fn>());
+    }
+
+    template <class _Fn>
+    _NODISCARD _Fn* _Large_fn_ptr() const noexcept {
+        return static_cast<_Fn*>(_Pointers[1]);
+    }
+
+    void _Set_large_fn_ptr(void* const _Value) noexcept {
+        _Pointers[1] = _Value;
+    }
+};
+
+// Size of a large function. Treat an empty function as if it has this size.
+// Treat a small function as if it has this size too if it fits and is trivially copyable.
+inline constexpr size_t _Minimum_function_size = 2 * sizeof(void*);
+
+template <class _Rx, class... _Types>
+[[noreturn]] _Rx __stdcall _Function_not_callable(const _Move_only_function_data&, _Types&&...) noexcept {
+    _CSTD abort(); // Unlike std::function, move_only_function doesn't throw bad_function_call
+                   // (N4901 [func.wrap.move.inv]/2)
+}
+
+template <class _Vt, class _VtInvQuals, class _Rx, bool _Noex, class... _Types>
+_NODISCARD _Rx __stdcall _Function_inv_small(const _Move_only_function_data& _Self, _Types&&... _Args) noexcept(_Noex) {
+    return _Invoker_ret<_Rx>::_Call(
+        static_cast<_VtInvQuals>(*_Self._Small_fn_ptr<_Vt>()), _STD forward<_Types>(_Args)...);
+}
+
+template <class _Vt, class _VtInvQuals, class _Rx, bool _Noex, class... _Types>
+_NODISCARD _Rx __stdcall _Function_inv_large(const _Move_only_function_data& _Self, _Types&&... _Args) noexcept(_Noex) {
+    return _Invoker_ret<_Rx>::_Call(
+        static_cast<_VtInvQuals>(*_Self._Large_fn_ptr<_Vt>()), _STD forward<_Types>(_Args)...);

```

Comment:
I support this, and I'd like macroizing forwarding via `static_cast`. But I prefer consistent solution in scope of #1208 instead of doing it here once

Reply:
I agree with consistency here (i.e. using `forward` unless and until we decide otherwise). @xiangfan-ms recently informed us that there isn't really a concern for compiler front-end throughput, so debug codegen size is the only remaining concern.